### PR TITLE
Decidir Plus: Add supported methods

### PR DESCRIPTION
**Pull request by ajawadmirza**
Added `authorize`, `capture`, `void`, `verify` methods for decidir plus
along with its test cases.

CE-2159

Remote:
15 tests, 51 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Unit:
5059 tests, 75068 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Rubocop:
728 files inspected, no offenses detected

### CODE REVIEW

File: lib/active_merchant/billing/gateways/decidir_plus.rb [hunk 1]
```diff
@@ -66,8 +95,8 @@ def add_payment(post, payment, options = {})
           post[:bin] = bin
         else
           post[:card_number] = payment.number
-          post[:card_expiration_month] = payment.month.to_s.rjust(2, '0')
-          post[:card_expiration_year] = payment.year.to_s[-2..-1]
+          post[:card_expiration_month] = format(payment.month, :two_digits)
+          post[:card_expiration_year] = format(payment.year, :two_digits)
```
Comment (naashton) [inline lib/active_merchant/billing/gateways/decidir_plus.rb:99]: 👍  I didn't know about this helper method

**Comment by naashton** [comment]
All of this looks good, but the remote tests are blowing up over some missing parameters in the request body.

```
       {"error_type"=>"invalid_request_error",
        "validation_errors"=>[{"code"=>"param_required", "param"=>"card_data"}]},
```

**Review by jessiagee — CHANGES_REQUESTED** [review]

**Comment by ajawadmirza** [comment]
> All of this looks good, but the remote tests are blowing up over some missing parameters in the request body.
> 
> ```
>        {"error_type"=>"invalid_request_error",
>         "validation_errors"=>[{"code"=>"param_required", "param"=>"card_data"}]},
> ```



> All of this looks good, but the remote tests are blowing up over some missing parameters in the request body.
> 
> ```
>        {"error_type"=>"invalid_request_error",
>         "validation_errors"=>[{"code"=>"param_required", "param"=>"card_data"}]},
> ```

@naashton I have shared the updated fixtures, let me know if it still doesn't work?

File: test/remote/gateways/remote_decidir_plus_test.rb [hunk 2]
```diff
@@ -3,7 +3,8 @@
 
 class RemoteDecidirPlusTest < Test::Unit::TestCase
   def setup
-    @gateway = DecidirPlusGateway.new(fixtures(:decidir_plus))
+    @gateway_purchase = DecidirPlusGateway.new(fixtures(:decidir_plus_purchase))
```
[the remark this answers is not in the document]
Comment (ajawadmirza) [reply test/remote/gateways/remote_decidir_plus_test.rb]: updated, please check :)

File: lib/active_merchant/billing/gateways/decidir_plus.rb [hunk 3]
```diff
@@ -106,10 +135,17 @@ def add_fraud_detection(post, options)
...
           hsh[:channel] = fraud_detection[:channel] if fraud_detection[:channel]
           hsh[:dispatch_method] = fraud_detection[:dispatch_method] if fraud_detection[:dispatch_method]
-          hsh[:csmdds] = fraud_detection[:csmdds] if fraud_detection[:csmdds]
+          csmdds_arr = []
+          fraud_detection[:csmdds].each do |csmdds|
+            csmdds_hsh = {}
+            csmdds_hsh[:code] = csmdds[:code].to_i
+            csmdds_hsh[:description] = csmdds[:description]
+            csmdds_arr.append(csmdds_hsh)
+          end
```
Comment (naashton) [inline lib/active_merchant/billing/gateways/decidir_plus.rb:147]: This code will result in an exception if `:csmdds` isn't set in the `options[:fraud_detection]` hash. I think we should break this code out into it's own method and you can call it with a conditional, `if fraud_detection[:csmdds]` or return early from the method.

**Review by naashton — CHANGES_REQUESTED** [review]
On the `csmdds` note, it might be worth adding a test case for when that field is not set, regardless of whether the request passes or fails.

**Review by jessiagee — APPROVED** [review]
Adding my 👍🏻 now just waiting for you to address @naashton 's points then should be g2g!

File: lib/active_merchant/billing/gateways/decidir_plus.rb (hunk 3 above)
Comment (ajawadmirza) [reply lib/active_merchant/billing/gateways/decidir_plus.rb]: Thanks, change is updated with unit test case as well.

**Comment by naashton** [comment]
I'm seeing 7 test failures, mostly with this error response:

```
        {"type"=>"processing_error",
         "reason"=>{"id"=>-1, "description"=>"", "additional_description"=>""}}}
```

Are you guys not seeing these?

**Pull request merged**